templates/concourse-worker-watchdog.j2

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+watchdog() {
+    #WORKER_PID=$1
+
+    while(true); do
+        FAIL=0
+
+        curl --silent 127.0.0.1:8888 || FAIL=1
+
+        #if [[ $FAIL -eq 0 ]]; then
+        if [[ $FAIL -eq 1 ]]; then
+            #/bin/systemd-notify --pid=$WORKER_PID "WATCHDOG=1";
+            /bin/systemctl restart concourse-worker
+            sleep 1
```

Comment:
I think the `sleep` should be bigger here to give time to the concourse-worker to clean his mess before being ready.  
With a `sleep 1` we might up in situations where the watchdog restarts the concourse-worker in a loop.

The logic might need to be different than the systemd watchdog implementation which allowed the healthcheck to fail several times before deciding to restart the concourse-worker.  
Here restarting at the first healthcheck failure is brutal when sometimes it might be a false alarm.  
Ideally we would want a to retry 2 or 3 times the healthcheck to be sure the concourse-worker is KO, restart the process if it is then wait some time for the concourse-worker to recover/clean up old things before expecting the healthcheck to be successful.  
  
WDYT ?

Reply:
Not sure for the sleep it should be good just after a worker restart but I added 3 retry (with 15sec between each) to be sure